tests/Fixer/ArrayNotation/MultilineLongArrayFixerTest.php

```diff
@@ -0,0 +1,437 @@
+                // Multiple nested arrays with max_length
+                <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => ['baz' => ['foo']],
+            ['baj']
+            ];
+            EXPECTED,
+                <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => ['baz' => ['foo']], ['baj']];
+            INPUT,
+                ['max_length' => 15],
+            ],
+            [
+                // Nested arrays 2
+                <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => [
+            'baz' => [
+            'foo'
+            ]
+            ],
+            ];
+            EXPECTED,
+                <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => ['baz' => ['foo']],];
+            INPUT,
+            ],
+            [
+                // Single line array with brackets inside of a string
+                <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => 'foo is [baz]',
+            ];
+            EXPECTED,
+                <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => 'foo is [baz]',];
+            INPUT,
+            ],
+            [
+                // Single line array with arrow function
+                <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => fn($i) => 'baz'
+            ];
+            EXPECTED,
+                <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => fn($i) => 'baz'];
+            INPUT,
+            ],
+            [
+                // Single line array with anonymous function
+                <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => function( $x ,$y) { return $x + $y; }
+            ];
+            EXPECTED,
+                <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => function( $x ,$y) { return $x + $y; }];
+            INPUT,
+            ],
+            [
+                // Internal short arrays inside long one.
+                <<<'EXPECTED'
+            <?php
+            $foo = [
+                'foo' => ['short' => $array,],
+                'bar' => ['other' => 'short','array' => 3,],
+            ];
+            EXPECTED,
+                <<<'INPUT'
+            <?php
+            $foo = [
+                'foo' => [
+                    'short' => $array,
+                ],
+                'bar' => [
+                    'other' => 'short',
+                    'array' => 3,
+                ],
+            ];
+            INPUT,
+                ['max_length' => 40],
+            ],
+            // don't change anonymous class implements list but change array inside
+            [
+                <<<'EXPECTED'
+            <?php
+            $x = [
+            1,
+            "2",
+            "c" => new class implements Foo ,Bar { const FOO = [
```

Comment:
```suggestion
            "c" => new class implements Foo, Bar { const FOO = [
```

Comma should be placed differently 😉, but this is only side comment - I am wondering if arrays like here should be refactored 🤔.

Reply:
Nice catch. As for your second comment, it's a good point. We could skip one-line stuff like classes, functions, arrow functions, conditions, ternary ?

Reply:
It's always safer to start with smaller range and then add new cases rather than introduce something that later needs to be maintained for backward compatibility. I would like to know what others think about it: @kubawerlos, @keradus, @julienfalque?

Reply:
I'm not sure what you mean, should we format everything (like how it is now) or should we skip as many special cases as we can ?

Reply:
I mean we should discuss it properly now and decide if test cases provided by you are OK because if we merge it, and people will start using, it will be harder to change behavior later 🙂. So if we decide that it's better to not touch some of these structures, then fixer has to be updated before merging.